packages/labs/src/select.tsx

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import * as classNames from "classnames";
+import * as PureRender from "pure-render-decorator";
+import * as React from "react";
+
+import {
+    AbstractComponent,
+    Button,
+    Classes,
+    HTMLInputProps,
+    IInputGroupProps,
+    InputGroup,
+    IPopoverProps,
+    Keys,
+    Popover,
+    Position,
+    Utils,
+} from "@blueprintjs/core";
+import { IInputListRendererProps, IListItemsProps, InputList } from "./inputList";
+
+export interface ISelectProps<T> extends IListItemsProps<T> {
+    /**
+     * Whether the dropdown list can be filtered.
+     * Disabling this option will remove the `InputGroup` and ignore `inputProps`.
+     * @default true
+     */
+    filterable?: boolean;
+
+    /**
+     * Custom renderer for an item in the dropdown list. Receives a boolean indicating whether
+     * this item is active (selected by keyboard arrows) and an `onClick` event handler that
+     * should be attached to the returned element.
+     */
+    itemRenderer: (item: T, isActive: boolean, onClick: React.MouseEventHandler<HTMLElement>) => JSX.Element;
+
+    /** React child to render when filtering items returns zero results. */
+    noResults?: string | JSX.Element;
```

Comment:
thoughts on changing this to `nonIdealState`? I think that this could be used for displaying a spinner if the user decides to use server-side search/pagination on input change